inference-engine/src/transformations/src/transformations/serialize.cpp

```diff
@@ -298,6 +299,60 @@ bool is_exec_graph(const ngraph::Function& f) {
     return false;
 }
 
+void resolve_dynamic_shapes(const ngraph::Function& f) {
+    auto f_clone = ngraph::clone_function(f);
+
+    const auto & f_ops = f.get_ordered_ops();
+    const auto & f_clone_ops = f_clone->get_ordered_ops();
+    assert(f_ops.size() == f_clone_ops.size());
+
+    for (size_t id = 0; id < f_ops.size(); ++id) {
+        auto & op = f_ops[id];
+        auto & clone_op = f_clone_ops[id];
+
+        if (auto op_subgraph = std::dynamic_pointer_cast<op::util::SubGraphOp>(op)) {
+            resolve_dynamic_shapes(*op_subgraph->get_function());
+        }
+
+        op->validate_and_infer_types();
+        clone_op->validate_and_infer_types();
+
+        auto dynamic_to_static = [](const PartialShape & shape) -> PartialShape {
+            if (shape.is_static() || shape.rank().is_dynamic()) {
```

Comment:
How does it work if rank is dynamic? In any case we return the dynamic shape, how do we add such shapes in IR?